packages/cactus-plugin-ledger-connector-besu/src/main/typescript/plugin-ledger-connector-besu.ts

```diff
@@ -912,6 +913,77 @@ export class PluginLedgerConnectorBesu
     return deployResponse;
   }
 
+  public async deployContractNoKeychain(
+    req: DeployContractSolidityBytecodeNoKeychainV1Request,
+  ): Promise<DeployContractSolidityBytecodeV1Response> {
+    const fnTag = `${this.className}#deployContract()`;
+    Checks.truthy(req, `${fnTag} req`);
+    if (isWeb3SigningCredentialNone(req.web3SigningCredential)) {
+      throw new Error(`${fnTag} Cannot deploy contract with pre-signed TX`);
+    }
+    const { contractName, contractJSONString } = req;
+    const networkId = await this.web3.eth.net.getId();
+
+    const tmpContract = new this.web3.eth.Contract(req.contractAbi);
+    const deployment = tmpContract.deploy({
+      data: req.bytecode,
+      arguments: req.constructorArgs,
+    });
+
+    const abi = deployment.encodeABI();
+    const data = abi.startsWith("0x") ? abi : `0x${abi}`;
+    this.log.debug(`Deploying "${req.contractName}" with data %o`, data);
+
+    const web3SigningCredential = req.web3SigningCredential as
+      | Web3SigningCredentialPrivateKeyHex
+      | Web3SigningCredentialCactusKeychainRef;
+
+    const runTxResponse = await this.transact({
+      transactionConfig: {
+        data,
+        from: web3SigningCredential.ethAccount,
+        gas: req.gas,
+        gasPrice: req.gasPrice,
+      },
+      consistencyStrategy: {
+        blockConfirmations: 0,
+        receiptType: ReceiptType.NodeTxPoolAck,
+        timeoutMs: req.timeoutMs || 60000,
+      },
+      web3SigningCredential,
+      privateTransactionConfig: req.privateTransactionConfig,
+    });
+
+    const { transactionReceipt: receipt } = runTxResponse;
+    const { status, contractAddress } = receipt;
+
+    if (status && contractAddress) {
+      if (contractJSONString) {
+        const networkInfo = { address: contractAddress };
+        const contractJSON = JSON.parse(contractJSONString);
+        this.log.debug("Contract JSON: \n%o", JSON.stringify(contractJSON));
+        const contract = new this.web3.eth.Contract(
+          contractJSON.abi,
+          contractAddress,
+        );
+        this.contracts[contractName] = contract;
+        const network = { [networkId]: networkInfo };
+        contractJSON.networks = network;
+      } else {
+        throw new Error(
```

Comment:
@jagpreetsinghsasan I've got a new requirement here in terms of error handling, sorry for not saying this earlier. I wrote a little essay about it in another PR review and it's quite long and I'm tryin to save time so I'll just copy paste the relevant parts of that one here but if it's not immediately obvious how it applies then please let me know and I'll just go the whole way and re-write the text more clearly.

---

Exception throws and re-throws are done with the appropriate utilities which are either 
* `createRuntimeErrorWithCause()` from the file at `packages/cactus-common/src/main/typescript/exception/create-runtime-error-with-cause.ts` or
* The `http-errors` library's function collection accessed via `createHttpError` .  For HTTP REST API endpoint related error handling where your exceptions communicate with the web layer, have a look at the patterns established in this diff 
https://github.com/hyperledger/cacti/pull/2869
<br>In short, instead of this:
  ```typescript
  // This is wrong - the HTTP response's status code will be
  // set to 500 but it should be 400 because it was user error
  // not a bug in the server's production code.
  throw new Error(
  `${fnTag} Cannot deploy contract without keychainId and the contractName`
	);
	```
	Do this:
	```typescript
	if (!keychainHasContract) {
	// This is better, because the HTTP response will 
	// contain a status code of 400 instead of 500
	const errorMessage =
		`${fnTag} Cannot create an instance of the contract instance because` +
		`the contractName in the request does not exist on the keychain`;
	throw new createHttpError[400](errorMessage);
	}
	```